Components/Core/Timer.cpp

```diff
@@ -1,34 +1,226 @@
+{
+	if (xTimerDelete(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD*2) == pdPASS) {
+		SOAR_PRINT("Timer has been deleted \n\n");
+	}
+}
+
+/**
+ * @brief Callback function for polling timers
+*/
+void Timer::CallbackFunction(TimerHandle_t xTimer){
+	Timer* ptrTimer = (Timer*)pvTimerGetTimerID(xTimer);
+	ptrTimer->timerState = COMPLETE;
 }
 
 /**
- * @brief Changes this timer object's RTOS timer period, returns true on success, returns false on failure (timer command queue full)
+ * @brief Changes timer period, Sets timer state back to uninitialized and stops timer
  */
 bool Timer::ChangePeriod(const uint32_t period_ms)
 {
-	if (xTimerChangePeriod(rtTimerHandle, MS_TO_TICKS(period_ms), DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdTRUE)
+	if (xTimerChangePeriod(rtTimerHandle, MS_TO_TICKS(period_ms), DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdTRUE) {
+		if (xTimerStop(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdPASS) {
+			timerState = UNINITIALIZED;
+			return true;
+		}
+	}
+	return false;
+}
+
+/**
+ * @brief Changes timer period, Sets timer state back to counting and starts timer
+ */
+bool Timer::ChangePeriodAndStart(const uint32_t period_ms)
+{
+	if (xTimerChangePeriod(rtTimerHandle, MS_TO_TICKS(period_ms), DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdTRUE) {
+		timerState = COUNTING;
+		return true;
+	}
+	return false;
+}
+
+/**
+ * @brief Starts the timer
+*/
+bool Timer::Start()
+{
+	if ((timerState == COMPLETE) || (timerState == COUNTING)) {
+		return false;
+	}
+	else if (timerState == PAUSED) {
+		ChangePeriod(remainingTimeBetweenPauses);
+	}
+	if (xTimerStart(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdPASS) {
+		timerState = COUNTING;
 		return true;
+	}
 	return false;
 }
+
+/**
+ * @brief Stops the timer
+*/
+bool Timer::Stop()
+{
+	// Checks if timer is in counting state because it cannot be stopped in any other state
+	if (timerState != COUNTING) {
+		return false;
+	}
+	// Calculates the time left on the timer before it is paused
+	remainingTimeBetweenPauses = rtosTimeRemaning();
+	if (xTimerStop(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdPASS) {
+		timerState = PAUSED;
+		return true;
+	}
+	return false;
+}
+
+/**
+ * @brief Restarts timer without starting to count
+*/
+bool Timer::ResetTimer()
+{
+	if (timerState == UNINITIALIZED) {
+		return false;
+	}
+	if (xTimerReset(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdPASS) {
+		if (xTimerStop(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdPASS) {
+			timerState = UNINITIALIZED;
+			return true;
+		}
+	}
+	return false;
+}
+
+/**
+ * @brief Restarts Timer and starts counting
+*/
+bool Timer::ResetTimerAndStart()
+{
+	if (timerState == UNINITIALIZED) {
+		return false;
+	}
+	if (xTimerReset(rtTimerHandle, DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdPASS) {
+		timerState = COUNTING;
+		return true;
+	}
+	return false;
+}
+
+/**
+ * @brief Sets timer to auto-reload if parameter is set to true, Sets timer to one shot if parameter is set to false
```

Comment:
Please add an @param setReloadOn something like "auto reload to on if true, auto reload off if false"